pkg/compact/compact_e2e_test.go

```diff
@@ -416,3 +417,101 @@ func createAndUpload(t testing.TB, bkt objstore.Bucket, blocks []blockgenSpec) (
 	}
 	return metas
 }
+
+func Test_Issue2459_e2e(t *testing.T) {
```

Comment:
So what we do usually is just normal Test name and comment that `// Regression test for #xx issue`

Reply:
Wonder if we can embed this into some case for `fetcher_test` rather. What do you think? (: Just simpler.

Reply:
Maybe it was found, but it's best to test the contract of the filter not how we mitigated it outside (:  Not a blocker.